bson/value_reader.go

```diff
@@ -700,7 +687,7 @@ func (vr *valueReader) ReadElement() (string, ValueReader, error) {
 			return "", nil, vr.invalidDocumentLengthError()
 		}
 
-		vr.pop()
+		_ = vr.pop()
```

Comment:
We should handle this error.

Reply:
This line is inside an error-handling block. We don't need to overwrite the original error even when `pop()` returns an error.

Reply:
`ErrEOD` is a sentinel error, similar to `io.EOF`, so it's not an error handling block in the sense that "something has gone wrong". However, it does seem like the `pop` call here will never read from the underlying reader, so it should be safe to discard. Consider adding a comment that says the `pop` call here will never read from the underlying reader, so it's safe to discard the error.